homeassistant/components/plugwise/models.py

```diff
@@ -0,0 +1,35 @@
+"""Models for the Plugwise integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Generic, TypeVar, overload
+
+from plugwise import DeviceData
+
+from homeassistant.helpers.entity import EntityDescription
+
+T = TypeVar("T", bound=DeviceData)
+
+
+@dataclass
+class PlugwiseBaseMixin(EntityDescription, Generic[T]):
+    """Mixin for Plugwise."""
+
+    pw_key: str | None = None
+    pw_lookup: str | None = "sensors"
+
+    @overload
+    def pw_get_value(self, obj: T, ret: bool = False) -> bool:
+        ...
+
+    @overload
+    def pw_get_value(self, obj: T, ret: float | int = 0) -> float | int:
+        ...
+
+    def pw_get_value(self, obj: T, ret: Any = None) -> Any:
+        """Return value from Plugwise device."""
+        if result := obj[self.pw_lookup].get(self.pw_key):
+            return result  # pragma: no cover  # will be covered on sensor domain
```

Comment:
Lets remove the pragma comment here.

Reply:
As indicated in reply to epenet above here, I'm all up for cleaning up and re-introducing when needed - it was to show what we elude to with the upcoming PR (referring to the @overrides as well). If Core is content with the route to take I'll happily take the overrides and pragma out of model and reintroduce in a follow-up for sensor?

Reply:
Cleaned up accordingly in [76b1ef7](https://github.com/home-assistant/core/pull/93162/commits/76b1ef7c995beff4ec7bd524cca9904437743f57)